From 443c82679edd59ad7ec922a4f55e6625600a2496 Mon Sep 17 00:00:00 2001
From: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Date: Sun, 22 Jan 2012 16:22:24 -0500
Subject: [PATCH 355/625] Fix init bug

Fix reversed logic bug introduced by commit
80996790fa42072d6525f2adb9a910312077bef6

Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Signed-off-by: Jianchuan Wang <jianchuan.wang@windriver.com>
---
 drivers/staging/lttng2/lttng-abi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/staging/lttng2/lttng-abi.c b/drivers/staging/lttng2/lttng-abi.c
index d15e3f8..2b67377 100644
--- a/drivers/staging/lttng2/lttng-abi.c
+++ b/drivers/staging/lttng2/lttng-abi.c
@@ -752,7 +752,7 @@ int __init lttng_abi_init(void)
 	lttng_proc_dentry = proc_create_data("lttng", S_IWUSR, NULL,
 					&lttng_fops, NULL);
 	
-	if (lttng_proc_dentry) {
+	if (!lttng_proc_dentry) {
 		printk(KERN_ERR "Error creating LTTng control file\n");
 		ret = -ENOMEM;
 		goto error;
-- 
2.0.2

